### PR TITLE
Feat/striate installation

### PR DESCRIPTION
**Pull request by IanAtCredo**
## Describe your changes
Added the ability to install core functionality separate from heavier functionality. Please test various installations and missing requirements!

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have built basic tests for new functionality (particularly new evaluators)
- [ ] If new libraries have been added, I have checked that [readthedocs](https://readthedocs.org/projects/credoai-lens/) API documentation is constructed correctly
- [ ] Will this be part of a major product update? If yes, please write one phrase about this update.

## Extra-mile Checklist
- [ ] I have thought expansively about edge cases and written tests for them

### CODE REVIEW

**Review by credo-nate — APPROVED** [review]
`test.yml` looks good.

File: requirements.txt [hunk 1]
```diff
@@ -10,31 +10,13 @@ python-dotenv>=0.19.2
...
 # required for ShapExplainer evaluator
 shap>=0.41.0
 
-# required for Deepchecks evaluator
-deepchecks>=0.9.1
-
 # required for RankingFairness evaluator
 finsfairauditing>=0.0.2
 
-# required for NLP evaluator
```
Comment (esherman-credo) [inline requirements.txt:35]: These are fully cut? Seems like they're not in the extras file

File: credoai/evaluators/utils/utils.py [hunk 2]
```diff
@@ -15,3 +19,34 @@ def all_subclasses(cls):
     return set(cls.__subclasses__()).union(
         [s for c in cls.__subclasses__() for s in all_subclasses(c)]
     )
+
+
+def list_evaluators_exhaustive():
+    """List all defined evaluation modules"""
+    evaluator_path = os.path.dirname(os.path.dirname(__file__))
```
Comment (esherman-credo) [inline credoai/evaluators/utils/utils.py:26]: Will we ever have a case where evaluators are organized into further subfolders?

If so, this would need to be changed (heads up, but don't think it's a present problem to worry about)

File: setup.py [hunk 3]
```diff
@@ -31,8 +34,14 @@
     doc_requirements = [line.strip() for line in f]
 
 dev_requirements += doc_requirements
+full_requirements = extras_requirements
```
Comment (esherman-credo) [inline setup.py:37]: So my understanding is full = base + extra, all = base + extra + dev. The naming is a bit confusing since "full" and "all" could both be seen as meaning everything.

Maybe "docs", "extended", and "full"? Others for "full" could be "beta" or "experimental".

**Review by esherman-credo — CHANGES_REQUESTED** [review]
The files look good on review.
I am getting the following error on a cursory import of Lens.
Environment is fresh install with requirements.txt (the "base" version of Lens) --> seems we'll need to refactor the Keras validation checking in `model_utils.py` to be elsewhere/only get imported if TF/Lens-full is installed (those functions won't get called unless someone is using a Keras model so they'd have the necessary installations anyway)


image

File: requirements.txt (hunk 1 above)
Comment (IanAtCredo) [reply requirements.txt:35]: Yes. Until we bring the NLP evaluator back.

File: credoai/evaluators/utils/utils.py (hunk 2 above)
Comment (IanAtCredo) [reply credoai/evaluators/utils/utils.py:26]: I think this actually goes into subfolders, but yeah, this is based on our current organization.

File: setup.py (hunk 3 above)
Comment (IanAtCredo) [reply setup.py:37]: I just removed the "all". You can install both dev and extras using credoai_lens[full,dev]

**Review by amrasekh — APPROVED** [review]
Reviewed the changes and tested installation of core on fresh kernel and all worked as expected.

File: credoai/utils/model_utils.py [hunk 4]
```diff
@@ -1,11 +1,15 @@
 import warnings
-from credoai.utils import global_logger
 
 from sklearn.base import is_classifier, is_regressor
 from sklearn.ensemble import RandomForestClassifier
 from sklearn.utils import multiclass
 
-from tensorflow.keras import layers
+from credoai.utils import global_logger
```
Comment (esherman-credo) [inline credoai/utils/model_utils.py:7]: I don't think this file needs the "if tf_exists" paradigm in the `validate_keras_clf` function since we'd only be calling that function if keras is successfully imported elsewhere. That said, it might be good to put it in anyway.

**Review by esherman-credo — APPROVED** [review]
